pkg/azure/blob.go

```diff
@@ -343,6 +355,34 @@ func sharedCredentialFromSecret(endpoint string, secret *corev1.Secret) (*azblob
 	return nil, nil
 }
 
+// sasTokenFromSecret retrieves the SAS Token from the `sasKey`. It returns an empty string if the Secret
+// does not contain a valid set of credentials.
+func sasTokenFromSecret(ep string, secret *corev1.Secret) (string, error) {
+	if sasKey, hasSASKey := secret.Data[sasKeyField]; hasSASKey {
+		queryString := strings.TrimPrefix(string(sasKey), "?")
+		values, err := url.ParseQuery(queryString)
+		if err != nil {
+			return "", fmt.Errorf("unable to parse SAS TOKEN: %s", err)
+		}
+		epURL, err := url.Parse(ep)
+		if err != nil {
+			return "", fmt.Errorf("unable to parse endpoint url: %s", err)
```

Comment:
Uppercase to align with other parts of the code that refers to URLs.

```suggestion
			return "", fmt.Errorf("unable to parse endpoint URL: %s", err)
```